src/loki/isochrone_action.cc

```diff
@@ -31,6 +31,16 @@ void check_distance(const google::protobuf::RepeatedPtrField<valhalla::Location>
     }
   }
 }
+
+const std::tuple<float, float>
+get_max_metrics(const google::protobuf::RepeatedPtrField<valhalla::Contour>& contours) {
+  float max_dist, max_time = 0.0f;
+  for (auto& contour : contours) {
+    max_time = std::max(max_time, contour.time());
+    max_dist = std::max(max_dist, contour.distance());
```

Comment:
i think a contour should have distance or time but not both.

Reply:
If you go further in the review, you'll see that a request can have only time or distance. But when parsing it I found it more elegant to add 0 time or distance to the PBF object.

Reply:
you can and totally should have both in the same object that makes perfect sense but imho its way more standard in protobuf2 to check which one is set using `contour.has_time` or `contour .has_distance`

Reply:
can change that, sure. then will it return false if `contour.time() == 0`?

Reply:
sorry im suggesting this specifically:

```c++
max_time = std::max(max_time, contour.has_time() ? contour.time() : max_time);
max_dist = std::max(max_dist, contour.has_distance() ? contour.distance() : max_dist);
```

then you dont to to zero out the time if its missing or the distance if its missing in parsing code in worker.cc

Reply:
done